SUMO-282477 | Apoorv | Add. monitors to migrated v2 apps and small fixes (#6618)
SumoLogic/sumologic-documentation: docs/integrations/amazon-aws/elastic-container-service-container-insights-cloudwatch.md

| fields account
```

## Collect Container Insights performance log events for Task and Container

> Container Insights collects data asperformance log events using [embedded metric format](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/CloudWatch_Embedded_Metric_Format.html). More details [here](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/ContainerInsights.html).

Reviewer: ```suggestion
Container Insights collects data as performance log events using [embedded metric format](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/CloudWatch_Embedded_Metric_Format.html). More details [here](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/ContainerInsights.html).
```

## Collect Application Logs for Amazon ECS

Set up the Container logs collection using the steps in the following [docs](/docs/send-data/collect-from-other-data-sources/aws-fargate-log-collection). You can use AWS FireLens driver and avoid sending logs to CloudWatch log groups. Put account, region, and namespace fields also while configuring the source.

If your logs are already going to CloudWatch logs groups then you can create a subscription filter to subscribe the log groups to the delivery stream created in the previous step.

:::note
Application logs do not contain regions. You have to configure a new Sumo Logic source for each region if you want to avoid creating multiple sources, then you will have to put the [X-SUMO-Fields](/docs/manage/fields#x-sumo-fields-http-header) header inside logConfiguration by creating a custom fluent bit image and specify a custom fluent bit configuration.

For more information, see [Create a custom Fluent Bit image](/docs/send-data/collect-from-other-data-sources/aws-fargate-log-collection).
:::

## Collect Traces for Amazon ECS

> To set upcollection for traces:

Reviewer: ```suggestion
To set up collection for traces:
```

</details>

> ### Samplequery

Reviewer: ```suggestion
### Sample query
```

| transpose row _timeslice column resource_type
```

## Installing the Amazon ECS app

> Now that you have set up acollectionfor Amazon ECS with Container Insights and CloudWatch, install the Sumo Logic app for Amazon ECS with Container Insights and CloudWatch to use the pre-configured searches anddashboardsthat provide visibility into your environment for real-time analysis of overall usage.

Reviewer: ```suggestion
Now that you have set up a collection for Amazon ECS with Container Insights and CloudWatch, install the Sumo Logic app for Amazon ECS with Container Insights and CloudWatch to use the pre-configured searches and dashboards that provide visibility into your environment for real-time analysis of overall usage.
```

import AppInstall from '../../reuse/apps/app-install-v2.md';

<AppInstall/>

## Amazon ECS CloudWatch dashboards

### Cluster Overview

The **Amazon ECS - Cluster Overview** dashboard provides a high-level view of the cluster's health along with details on the utilized resources.

Use this dashboard to:

- Monitor the memory and CPU utilization of your cluster.
- View abnormal read-write activity and network incoming-outgoing bytes.

> <img src='https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/Amazon-ECS-Container-Insights/Amazon-ECS-Cluster-Overview.png' alt="Amazon ECS - Cluster Overviewdashboard" style={{border: '1px solid gray'}} width="800" />

Reviewer: ```suggestion
<img src='https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/Amazon-ECS-Container-Insights/Amazon-ECS-Cluster-Overview.png' alt="Amazon ECS - Cluster Overview dashboard" style={{border: '1px solid gray'}} width="800" />
```

### Cluster Performance Monitoring

The **Amazon ECS - Cluster Performance Monitoring** dashboard provides detailed information on the performance of your cluster, which you can use to fine-tune your cluster.

Use this dashboard to:

- Identify patterns and outliers over time.
- Monitor the performance of your cluster and use linked dashboards to drill down further into the root cause.

> <img src='https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/Amazon-ECS-Container-Insights/Amazon-ECS-Cluster-Performance-Monitoring.png' alt="Amazon ECS - Cluster Performance Monitoringdashboard" style={{border: '1px solid gray'}} width="800" />

Reviewer: ```suggestion
<img src='https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/Amazon-ECS-Container-Insights/Amazon-ECS-Cluster-Performance-Monitoring.png' alt="Amazon ECS - Cluster Performance Monitoring dashboard" style={{border: '1px solid gray'}} width="800" />
```

<img src='https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/Amazon-ECS-Container-Insights/Amazon-ECS-Cluster-Resource-Reservation.png' alt="Amazon ECS - Cluster Resource Reservation dashboard" style={{border: '1px solid gray'}} width="800" />

### Container Logs

The **Amazon ECS - Container Logs** dashboard provides detailed information on what is happening (errors or recent events) in a container.

Use this dashboard to:

- View recent logs of your container.
- Identify common errors and abnormal spikes in errors.

<img src='https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/Amazon-ECS-Container-Insights/Amazon-ECS-Container-Logs.png' alt="Amazon ECS - Container Logs dashboard" style={{border: '1px solid gray'}} width="800" />

### Container Overview

The **Amazon ECS - Container Overview** dashboard provides a high-level view of the health of the container along with details on the utilized resources.

Use this dashboard to:

- Track the container status and identify the container details like its task definition, image, account, etc.
- Monitor CPU, memory, disk, and network activity of your container.

> <img src='https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/Amazon-ECS-Container-Insights/Amazon-ECS-Container-Overview.png' alt="Amazon ECS - Container Overviewdashboard" style={{border: '1px solid gray'}} width="800" />

Reviewer: ```suggestion
<img src='https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/Amazon-ECS-Container-Insights/Amazon-ECS-Container-Overview.png' alt="Amazon ECS - Container Overview dashboard" style={{border: '1px solid gray'}} width="800" />
```

### EC2 LaunchType

> The **Amazon ECS - EC2LaunchType** dashboard provides a high-level view of the health of the cluster along with details on the utilized resources for EC2 launch types.

Reviewer: ```suggestion
The **Amazon ECS - EC2 LaunchType** dashboard provides a high-level view of the health of the cluster along with details on the utilized resources for EC2 launch types.
```

Use this dashboard to:

- Monitor CPU and memory utilization of clusters with EC2 launch type.
- View the number of clusters and tasks with EC2 launch type.

> <img src='https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/Amazon-ECS-Container-Insights/Amazon-ECS-EC2-LaunchType.png' alt="Amazon ECS - EC2LaunchTypedashboard" style={{border: '1px solid gray'}} width="800" />

Reviewer: ```suggestion
<img src='https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/Amazon-ECS-Container-Insights/Amazon-ECS-EC2-LaunchType.png' alt="Amazon ECS - EC2 LaunchType dashboard" style={{border: '1px solid gray'}} width="800" />
```

### Service Overview

The **Amazon ECS - Service Overview** dashboard provides a high-level view of the health of the services along with details on the utilized resources.

Use this dashboard to:

- Monitor the number of running, desired, and pending tasks.
- Identify services with abnormal CPU, network, memory, and disk activity.

> <img src='https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/Amazon-ECS-Container-Insights/Amazon-ECS-Service-Overview.png' alt="Amazon ECS - Service Overviewdashboard" style={{border: '1px solid gray'}} width="800" />

Reviewer: ```suggestion
<img src='https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/Amazon-ECS-Container-Insights/Amazon-ECS-Service-Overview.png' alt="Amazon ECS - Service Overview dashboard" style={{border: '1px solid gray'}} width="800" />
```

### Service Performance Monitoring

The **Amazon ECS - Service Performance Monitoring** dashboard provides detailed information on the performance of your services which you can use to fine-tune your cluster.

Use this dashboard to:

- Identify patterns and outliers over time for each of the resource metrics like CPU memory network and disk.
- Track the running, pending, and desired tasks trend.
- Monitor the performance of your services and use linked dashboards to drill down further into the root cause.

> <img src='https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/Amazon-ECS-Container-Insights/Amazon-ECS-Service-Performance-Monitoring.png' alt="Amazon ECS - Service Performance Monitoringdashboard" style={{border: '1px solid gray'}} width="800" />

Reviewer: ```suggestion
<img src='https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/Amazon-ECS-Container-Insights/Amazon-ECS-Service-Performance-Monitoring.png' alt="Amazon ECS - Service Performance Monitoring dashboard" style={{border: '1px solid gray'}} width="800" />
```

### Tasks Definition Family Overview

The **Amazon ECS - Tasks Definition Family Overview** dashboard provides a high-level view of the health of the tasks belonging to a particular task definition family and details on the utilized resources.

Use this dashboard to:

- View the number of tasks running with a single task definition family.
- Monitor CPU and memory usage by task definition family.

> <img src='https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/Amazon-ECS-Container-Insights/Amazon-ECS-Tasks-Definition-Family-Overview.png' alt="Amazon ECS - Tasks Definition Family Overviewdashboard" style={{border: '1px solid gray'}} width="800" />

Reviewer: ```suggestion
<img src='https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/Amazon-ECS-Container-Insights/Amazon-ECS-Tasks-Definition-Family-Overview.png' alt="Amazon ECS - Tasks Definition Family Overview dashboard" style={{border: '1px solid gray'}} width="800" />
```

### Tasks Overview

The **Amazon ECS - Tasks Overview** dashboard provides a high-level view of the health of the task along with details on the utilized resources and where they are running.

Use this dashboard to:

- View details of all the task instances and their launch type.
- Track Network Errors and Dropped Packets
- Monitor CPU, memory, disk, and network performance by task instances.

> <img src='https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/Amazon-ECS-Container-Insights/Amazon-ECS-Tasks-Overview.png' alt="Amazon ECS - Tasks Overviewdashboard" style={{border: '1px solid gray'}} width="800" />

Reviewer: ```suggestion
<img src='https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/Amazon-ECS-Container-Insights/Amazon-ECS-Tasks-Overview.png' alt="Amazon ECS - Tasks Overview dashboard" style={{border: '1px solid gray'}} width="800" />
```

### Tasks Definition Family Performance Monitoring

The **Amazon ECS - Tasks Definition Family Performance Monitoring** dashboard provides detailed information on the performance of your tasks which you can use to fine-tune your cluster.

Use this dashboard to:

- Identify patterns and outliers over time for each of the resource metrics like CPU, memory, network, and disk.

> - Monitor the performance of your tasksand use linked dashboards to drill down further into the root cause.

Reviewer: ```suggestion
- Monitor the performance of your tasks and use linked dashboards to drill down further into the root cause.
```

> <img src='https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/Amazon-ECS-Container-Insights/Amazon-ECS-Tasks-Definition-Family-Performance-Monitoring.png' alt="Amazon ECS - Tasks Definition Family Performance Monitoringdashboard" style={{border: '1px solid gray'}} width="800" />

Reviewer: ```suggestion
<img src='https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/Amazon-ECS-Container-Insights/Amazon-ECS-Tasks-Definition-Family-Performance-Monitoring.png' alt="Amazon ECS - Tasks Definition Family Performance Monitoring dashboard" style={{border: '1px solid gray'}} width="800" />
```

### Task Definition Family Resource Reservation

The **Amazon ECS - Task Definition Family Resource Reservation** dashboard provides information on resource reservation which can be used to set the right resource limits at the task definition level.

Use this dashboard to:

- Identify the right limits for CPU and memory reservations.

> <img src='https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/Amazon-ECS-Container-Insights/Amazon-ECS-Task-Definition-Family-Resource-Reservation.png' alt="Amazon ECS - Task Definition Family Resource Reservationdashboard" style={{border: '1px solid gray'}} width="800" />

Reviewer: ```suggestion
<img src='https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/Amazon-ECS-Container-Insights/Amazon-ECS-Task-Definition-Family-Resource-Reservation.png' alt="Amazon ECS - Task Definition Family Resource Reservation dashboard" style={{border: '1px solid gray'}} width="800" />
```

### Fargate LaunchType

The **Amazon ECS - Fargate LaunchType** dashboard provides a high-level view of the cluster's health along with details on the utilized resources for Fargate launch types.

Use this dashboard to:

- Monitor network activity of your clusters with Fargate launch type.
- View the number of clusters and tasks with Fargate launch type.

> <img src='https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/Amazon-ECS-Container-Insights/Amazon-ECS-Fargate-LaunchType.png' alt="Amazon ECS - Fargate LaunchTypedashboard" style={{border: '1px solid gray'}} width="800" />

Reviewer: ```suggestion
<img src='https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/Amazon-ECS-Container-Insights/Amazon-ECS-Fargate-LaunchType.png' alt="Amazon ECS - Fargate LaunchType dashboard" style={{border: '1px solid gray'}} width="800" />
```

### Audit Events

The **Amazon ECS - Audit Events** dashboard gives information on the type of request made to ECS, the IP making the request, who made it and when, and more.

Use this dashboard to:

- View audit trail of actions taken by a user, role, or AWS service in Amazon ECS.

> - Monitor container registration/deregistrationevents.

Reviewer: ```suggestion
- Monitor container registration/deregistration events.
```

- Identify location, IP address from where the request was made, and resource crud events over time.

> <img src='https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/Amazon-ECS-Container-Insights/Amazon-ECS-Audit-Events.png' alt="Amazon ECS - Audit Eventsdashboard" style={{border: '1px solid gray'}} width="800" />

Reviewer: ```suggestion
<img src='https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/Amazon-ECS-Container-Insights/Amazon-ECS-Audit-Events.png' alt="Amazon ECS - Audit Events dashboard" style={{border: '1px solid gray'}} width="800" />
```
